activesupport/test/core_ext/time_ext_test.rb

```diff
@@ -149,6 +149,9 @@ def test_end_of_day
       assert_equal Time.local(2006,3,19,23,59,59,Rational(999999999, 1000)), Time.local(2006,3,19,10,10,10).end_of_day, 'ends DST'
       assert_equal Time.local(2006,10,1,23,59,59,Rational(999999999, 1000)), Time.local(2006,10,1,10,10,10).end_of_day, 'start DST'
     end
+    with_env_tz 'Asia/Yekaterinburg' do
+      assert_equal Time.local(2015, 2, 8, 23, 59, 59, Rational(999999999, 1000)), Time.new(2015, 2, 8, 8, 0, 0, '+05:00').end_of_day
+    end
```

Comment:
Indirectly. All methods like `Time#end_of_*` that use `Time#change` pass `usec: Rational(999999999, 1000)` as their parameter may be affected by this error.

For example:

```ruby
Time.new(2015, 5, 9, 10, 00, 00, '+03:00').end_of_day # raise ArgumentError: argument out of range
```